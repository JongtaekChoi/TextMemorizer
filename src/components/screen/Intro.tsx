import Button from '../shared/Button';
import { IC_MASK } from '../../utils/Icons';
import React from 'react';
import { RootStackNavigationProps } from '../navigation/RootStackNavigator';
import { User } from '../../types';
import { View } from 'react-native';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useAppContext } from '../../providers/AppProvider';
import { useThemeContext } from '@dooboo-ui/native-theme';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  overflow: scroll;
  background-color: ${({ theme }): string => theme.background};

  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
`;

const ContentWrapper = styled.View`
  flex-direction: column;
  flex: 1;
  width: 100%;
  justify-content: center;
  align-items: center;
`;

const ButtonWrapper = styled.View`
  flex-direction: column;
  bottom: 40px;
  width: 85%;
  align-self: center;
`;

const StyledText = styled.Text`
  font-size: 28px;
  line-height: 27px;
  color: ${({ theme }): string => theme.fontColor};
`;

interface Props {
  navigation: RootStackNavigationProps<'Intro'>;
}

function Intro(props: Props): React.ReactElement {
  const { changeThemeType } = useThemeContext();

  return (
    <Container>
      <ContentWrapper>
        <StyledText>Text Memorizer v1.0</StyledText>
      </ContentWrapper>
      <ButtonWrapper>
        <View style={{ marginTop: 8 }} />
        <Button
          testID="btn-navigate"
          onClick={(): void => props.navigation.navigate('TextList')}
          text={getString('NAVIGATE', { name: 'TextList' })}
        />
        <View style={{ marginTop: 8 }} />
        <Button
          testID="btn-theme"
          onClick={(): void => changeThemeType()}
          text={getString('CHANGE_THEME')}
        />
      </ButtonWrapper>
    </Container>
  );
}

export default Intro;
