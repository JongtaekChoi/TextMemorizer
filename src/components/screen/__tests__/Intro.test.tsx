import React, { ReactElement } from 'react';
import { RenderResult, act, fireEvent, render } from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Button from '../../shared/Button';
import Intro from '../Intro';
import { ThemeType } from '@dooboo-ui/native-theme';
import renderer from 'react-test-renderer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;
let testingLib: RenderResult;

describe('[Intro] screen rendering test', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Intro {...props} />);
    testingLib = render(component);
  });

  it('should render outer component and snapshot matches', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toBeTruthy();
  });

  it('should render [Dark] theme', () => {
    component = createTestElement(<Intro {...props} />, ThemeType.DARK);
    testingLib = render(component);
    const { baseElement } = testingLib;
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('[Intro] Interaction', () => {
  let testingLib: RenderResult;

  it('should navigate when button has clicked', () => {
    testingLib = render(component);

    act(() => {
      fireEvent.press(testingLib.getByTestId('btn-navigate'), 'click');
    });
    expect(props.navigation.navigate).toHaveBeenCalledWith('TextList');
  });

  it('should change theme when button has clicked', () => {
    testingLib = render(component);

    act(() => {
      fireEvent.press(testingLib.getByTestId('btn-theme'), 'click');
    });
  });
});
