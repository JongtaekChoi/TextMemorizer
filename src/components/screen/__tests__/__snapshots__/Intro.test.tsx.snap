// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Intro] screen rendering test should render [Dark] theme 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#323739",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-start",
            "overflow": "hidden",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 28,
                "lineHeight": 27,
              },
            ]
          }
        >
          Text Memorizer v1.0
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "bottom": 40,
              "flexDirection": "column",
              "width": "85%",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginTop": 8,
            }
          }
        />
        <TouchableOpacity
          activeOpacity={0.5}
          testID="btn-navigate"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#069ccd",
                  "borderColor": "#069ccd",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "height": 52,
                  "justifyContent": "center",
                  "width": 320,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Navigate to TextList
            </Text>
          </View>
        </TouchableOpacity>
        <View
          style={
            Object {
              "marginTop": 8,
            }
          }
        />
        <TouchableOpacity
          activeOpacity={0.5}
          testID="btn-theme"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#069ccd",
                  "borderColor": "#069ccd",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "height": 52,
                  "justifyContent": "center",
                  "width": 320,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Change theme
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render outer component and snapshot matches 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 28,
            "lineHeight": 27,
          },
        ]
      }
    >
      Text Memorizer v1.0
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "bottom": 40,
          "flexDirection": "column",
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "marginTop": 8,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      testID="btn-navigate"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#069ccd",
              "borderColor": "#069ccd",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": 52,
              "justifyContent": "center",
              "width": 320,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
              },
            ]
          }
        >
          Navigate to TextList
        </Text>
      </View>
    </TouchableOpacity>
    <View
      style={
        Object {
          "marginTop": 8,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      testID="btn-theme"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#069ccd",
              "borderColor": "#069ccd",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": 52,
              "justifyContent": "center",
              "width": 320,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
              },
            ]
          }
        >
          Change theme
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
